velox/functions/prestosql/tests/RandTest.cpp

```diff
@@ -74,5 +74,124 @@ TEST_F(RandTest, nonNullInt8) {
   EXPECT_LT(rand(4), 4);
 }
 
+class SecureRandTest : public functions::test::FunctionBaseTest {
+ protected:
+  template <typename T>
+  std::optional<T> secureRand(std::optional<T> lower, std::optional<T> upper) {
+    return evaluateOnce<T>("secure_random(c0, c1)", lower, upper);
+  };
+};
+
+TEST_F(SecureRandTest, zeroArg) {
+  auto result =
+      evaluateOnce<double>("secure_random()", makeRowVector(ROW({}), 1));
+  EXPECT_LT(result, 1.0);
+  EXPECT_GE(result, 0.0);
+
+  result = evaluateOnce<double>("secure_rand()", makeRowVector(ROW({}), 1));
+  EXPECT_LT(result, 1.0);
+  EXPECT_GE(result, 0.0);
+}
+
+TEST_F(SecureRandTest, int64Args) {
+  auto result =
```

Comment:
Please add test cases with kInf, kNan, numeric_limits::max(), numeric_limits::min() and null values for all the functions.